rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
The more I think about this, the more I think that no plugin work should happen during handling of the `new_view` rpc, and that adding a notification for when a view is ready to be interacted with is a useful and reasonable part of view lifecycle.

This simplifies code in a number of places, and makes the API more explicit; a client should not receive any updates for a given view until it has sent a 'view_loaded` (or whatever name) notification.

Because this notification already exists (by another name) I'm going to remove all of the `run_on_idle` stuff currently here, and implement this in terms of that notification, which will be trivial; we can reconsider from there.